irohad/network/impl/async_grpc_client.hpp

```diff
@@ -39,9 +38,7 @@ namespace iroha {
           if (not call->status.ok()) {
             log_->warn("RPC failed: {}", call->status.error_message());
           }
-          if (call->on_response) {
-            call->on_response(call->status, call->reply);
-          }
+          call->onResponse();
```

Comment:
It seems a good location to catch exceptions and write into log file. Or even proceed to work.

Reply:
```c++
std::unique_ptr<AsyncClientCall> call(static_cast<AsyncClientCall*>(got_tag));
// delete call;

Reply:
Fixed